### PR TITLE
silence r,c  warnings when property does not depend on r,c

### PR DESCRIPTION
**Pull request by holly-c-gibbs**
Addresses issue #3667 by filtering r,c warning for region properties that don't depend on ordering of r,c (eccentricity, major_axis_length, and minor_axis_length). 
## Checklist
- [x] Clean style in [the spirit of PEP8](https://www.python.org/dev/peps/pep-0008/)

## For reviewers

- [ ] Check that the PR title is short, concise, and will make sense 1 year
  later.
- [ ] Check that new functions are imported in corresponding `__init__.py`.
- [ ] Check that new features, API changes, and deprecations are mentioned in
      `doc/release/release_dev.rst`.
- [ ] Consider backporting the PR with `@meeseeksdev backport to v0.14.x`

### CODE REVIEW

File: skimage/measure/_regionprops.py [hunk 1]
```diff
@@ -194,7 +195,7 @@ def coords(self):
                           for i in range(self._ndim)]).T
 
     @only2d
-    def eccentricity(self):
+    def eccentricity(self): #suppress warning
```
Comment (jni) [inline skimage/measure/_regionprops.py]: @holly-c-gibbs I think you meant to use this comment as a to-do. The line underneath should also be silenced.

**Comment by jni** [comment]
@scikit-image/core in offline chats with @holly-c-gibbs, I initially thought it was a bit of a waste of time to silence the warnings since the whole xy machinery should go away in 0.16. However I think it's valuable to do it in two steps, and backport to 0.14 and 0.15, and then remove the whole thing altogether.

**Comment by hmaarrfk** [comment]
This seems like it should be addressed in `inertia_tensor_eigvals` no?

**Comment by jni** [comment]
@hmaarrfk Ach, sorry we were too fast. I kinda wondered about that. Are the eigvals indeed invariant to the coordinate convention?

**Comment by alexandrejaguar** [comment]
Thank you @holly-c-gibbs!

**Comment by hmaarrfk** [comment]
seems like that is all it is doing right? finding  when the eigenvalues are computed and going off that.

**Comment by hmaarrfk** [comment]
It is unclear if this is even helping the situation.

I'm not able to parse where the warning is comming from.
https://github.com/scikit-image/scikit-image/blob/v0.15.x/skimage/measure/_regionprops.py#L249

You all see to be in pysical proximity, but this behaviour change didn't come with any tests that would indicate that the underlying challenge is fixed. A test that uses: `with expected_warnings(None)` would have done well here.

**Comment by jni** [comment]
@hmaarrfk thanks for the review. Actually it seems the warning was already removed from master. :man_facepalming: Will fix, thanks for the help!

**Comment by hmaarrfk** [comment]
I would revert on master, and target the PR directly to the 0.14 branch. Personally I don't think 0.15.1 is happening.

If it is happening, we can likely cherrypick the commit from 0.14.x

**Comment by jni** [comment]
:+1:

**Comment by hmaarrfk** [comment]
If you want help, I can try and get things building on 0.15 again. But it is going to require significant backporting I think.

**Comment by jni** [comment]
don't stress. =)

**Comment by hmaarrfk** [comment]
THere isn't any stress, just trying to give constructive critisms as opposed to: "this isn't the right approach", and offering to help where I can.

**Comment by jni** [comment]
I'm happy to push on 0.16 in general. We want to get this Quiet Edition of skimage out ASAP! =)

**Comment by hmaarrfk** [comment]
yeha, but RC is a big change, and will likely break stuff.

**Comment by hmaarrfk** [comment]
no matter how hard we try not to.

**Pull request merged**